### ISSUE TITLE
Fragment Built-in Output Value for Stencil [Feature Request]

### ISSUE THREAD

**Issue by Gilberetzu** [gpuweb/gpuweb#2553]
Hello!,

I was reading through the WebGPU and WGSL specs, and it seems impossible to change the stencil reference value on the fragment stage. It seems to be possible on other rendering APIs, on D3D12 and D3D11.3 is specified here (https://docs.microsoft.com/en-us/windows/win32/direct3d12/shader-specified-stencil-reference-value), on Vulkan is defined here (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/FragStencilRefEXT.html), and it also seems to be one of the "Fragment Function Output Attributes" since Metal 2.1 (https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf). I hope this gets added at some point🤞, or if it is already supported, it would be awesome if someone could point me to where I can read about that😅.

Thank you so much in advance, and sorry if this was not supposed to be posted here (I am not entirely sure how I should file the feature request)

**Comment by dneto0** [comment]
Thanks for the report. (It's well formed and well phrased!)

**Comment by litherum** [comment]
Looks optional on Vulkan? That limits the appeal of doing this immediately.

**Comment by Gilberetzu** [comment]
Yes, that seems to be the case. `StencilExportEXT` is one of the capabilities that **_may_** _be supported in Vulkan implementations_ (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/chap46.html#spirvenv-capabilities), and as far as I can tell is the capability required to make the `StencilRefReplacingEXT` execution mode work (? not sure), which is required by the `FragStencilRefEXT`.

The Direct3D 12 docs are more explicit and state (https://docs.microsoft.com/en-us/windows/win32/direct3d12/shader-specified-stencil-reference-value):
> This feature is optional in both D3D12 and D3D11.3. To test for its support, check the PSSpecifiedStencilRefSupported boolean field of [D3D12_FEATURE_DATA_D3D12_OPTIONS](https://docs.microsoft.com/en-us/windows/desktop/api/d3d12/ns-d3d12-d3d12_feature_data_d3d12_options) using [CheckFeatureSupport](https://docs.microsoft.com/en-us/windows/desktop/api/d3d12/nf-d3d12-id3d12device-checkfeaturesupport).

But I completely understand why it would have a low priority, as it can be recreated pretty easily using an extra color attachment, plus I am sure there are a lot more important features not yet implemented. My first intention with this was to make sure I was not missing something pretty obvious (by misreading or not understanding the spec).
